Profile page integration

HAScrashed opened the pull request:
Please have mercy & good luck

HAScrashed committed:
Created edit password component

HAScrashed committed:
Create email component - TODO fix

Todo fix to static width + add loading circle

HAScrashed committed:
Added confirm new email component

HAScrashed committed:
Updated user service

HAScrashed committed:
Added missing translations

HAScrashed committed:
Updated Error handler service

HAScrashed committed:
Updated style of editable text

HAScrashed committed:
Updated profile-page

HAScrashed committed:
The ** route was blocking other routes, removed and added chage-email route

HAScrashed committed:
Minor changes

HAScrashed committed:
Minor change

HAScrashed committed:
Added loading circles for email and password change

HAScrashed committed:
Stopped unnecessary api fetch for profile page when possible

RobertPintera commented on src/app/components/features/profile-page/edit-password/edit-password.component.ts:
```diff
@@ -0,0 +1,101 @@
+  changePasswordFormHasErrors: Signal<boolean> = computed(() => {
+    return !this.repeatPassMatch() || this.hasPasswordErrors() || this.hasOldPasswordErrors();
+  });
+  private _oldPasswordErrors: WritableSignal<string[]> = signal([]);
```
I suggest using `signal<Type>()` instead of `WritableSignal<string[]>`. 
It’s shorter and more readable.

RobertPintera reviewed (approved):
I have one suggestion to use `signal()`, but I think the code is good.

HAScrashed committed:
Silenced the warnings of RobertPinteraCompiler™©

Changed some signals to readonly as the rest. Used different notation for signal declaration. I edited it in github site itself (in case something doesnt work now it's obvious why)
